packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -152,26 +154,44 @@ const CommonTokens = ({ onTokenSelect }: CommonTokensProps) => (
   </div>
 );
 
-const TokenListRow = ({ onTokenSelect, token }: TokenListRowProps) => (
-  <div
-    className="flex justify-between w-full py-2 cursor-pointer hover:bg-surface-50"
-    key={token.address}
-    onClick={() => onTokenSelect(token)}
-  >
-    <div className="flex items-center justify-between w-full">
-      <div className="flex items-center space-x-3">
-        <TokenIcon token={token} size="md" />
-        <div className="flex flex-col">
-          <BodyText size={2}>{token.symbol}</BodyText>
-          <BodyText className="text-em-low" size={1}>
-            {token.name}
-          </BodyText>
+const TokenListRow = ({ onTokenSelect, token }: TokenListRowProps) => {
+  const { chain } = useNetwork();
+  const { address } = useAccount();
+  const { data: balance } = useBalance({
+    address,
+    token: token?.address as `0x$string`,
+    chainId: chain?.id,
+  });
```

Comment:
I think we can push this for now, it was slow in the first run, but then it used the fetched values.